test/e2e/scalenodes.go

```diff
@@ -75,11 +81,38 @@ var _ = Describe("Scale nodes", func() {
 		err = waitForScale(mss.Items[0].Name)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	Specify("operator should maintain at least three worker replicas", func() {
+		ctx := context.Background()
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		mss, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mss.Items).NotTo(BeEmpty())
+
+		// Remove a replica from a worker MachineSet, and wait
+		err = scale(mss.Items[0].Name, -1)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = waitForScale(mss.Items[0].Name)
+		Expect(err).NotTo(HaveOccurred())
+
+		ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).Get(ctx, mss.Items[0].Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect MachineSet controller to have reverted the change, unless custom MachineSet
+		if strings.Contains(mss.Items[0].Name, instance.Spec.InfraID) {
+			Expect(ms.Status.Replicas).To(BeEquivalentTo(1))
+		} else {
+			Expect(ms.Status.Replicas).To(BeEquivalentTo(0))
+		}
```

Comment:
Right, I wanted to go ahead and keep the strategy already used in `scalenodes.go` (just list and scale), but now that we are making it mandatory 2 instead of 3 we would probably need multiple code paths in E2E to account for availability zone differences. I've removed E2E for now so that it passes, but maybe we can discuss this and `PrependReactor` off list